komodo-relational/src/main/java/org/komodo/relational/teiid/internal/TeiidImpl.java

```diff
@@ -881,121 +879,58 @@ public CachedTeiid importContent(UnitOfWork transaction) throws KException {
-        }
+        // Gets a teiid instance and connects if not connected
+        TeiidInstance teiidInstance = getConnectedTeiidInstance(transaction);
 
+        // Get the teiid instance artifacts and add them to CachedTeiid
         try {
-            teiidInstance.connect();
-            if (! teiidInstance.isConnected()) {
-                throw new KException(Messages.getString(Messages.Relational.TEIID_INSTANCE_CONNECTION_ERROR));
-            }
-
             Collection<TeiidVdb> instVdbs = teiidInstance.getVdbs();
-            if (instVdbs == null)
-                instVdbs = Collections.emptyList();
+            if (instVdbs == null) instVdbs = Collections.emptyList();
 
             Collection<TeiidDataSource> instDataSrcs = teiidInstance.getDataSources();
-            if (instDataSrcs == null)
-                instDataSrcs = Collections.emptyList();
+            if (instDataSrcs == null) instDataSrcs = Collections.emptyList();
 
             Collection<TeiidTranslator> instTranslators = teiidInstance.getTranslators();
-            if (instTranslators == null)
-                instTranslators = Collections.emptyList();
+            if (instTranslators == null) instTranslators = Collections.emptyList();
 
             Collection<String> instDataSourceTypeNames = teiidInstance.getDataSourceTypeNames();
-            if (instDataSourceTypeNames == null)
-                instDataSourceTypeNames = Collections.emptyList();
+            if (instDataSourceTypeNames == null) instDataSourceTypeNames = Collections.emptyList();
 
-            //
-            // Process the vdbs
-            //
+            // Add VDBs to CachedTeiid
             for (TeiidVdb teiidVdb : instVdbs) {
                 if (teiidVdb == null)
                     continue;
 
                 // Export the vdb content into a string
                 String content = teiidVdb.export();
-                if (content == null)
-                    continue;
-
                 if (StringUtils.isEmpty(content))
                     continue;
 
-                String vdbName = teiidVdb.getName();
-
-                // Output the content to a temp file
-                File tempFile = File.createTempFile(VDB_PREFIX, XML_SUFFIX);
-                Files.write(Paths.get(tempFile.getPath()), content.getBytes());
-
-                KomodoObject kobject = cachedTeiid.addChild(transaction, vdbName, VdbLexicon.Vdb.VIRTUAL_DATABASE);
-                Vdb vdb = new VdbImpl( transaction, getRepository(), kobject.getAbsolutePath());
-                vdb.setOriginalFilePath(transaction, tempFile.getAbsolutePath());
-                vdb.setVdbName( transaction, vdbName );
-
-                KomodoObject fileNode = vdb.addChild(transaction, JcrLexicon.CONTENT.getString(), null);
-                fileNode.setProperty(transaction, JcrLexicon.DATA.getString(), content);
+                // Add VDB to CachedTeiid
+                addOrReplaceCachedVdb(transaction, teiidVdb, cachedTeiid);
```

Comment:
What good is AddOrReplace methods if the RelationalModelFactory first wipes out the CachedTeiid and creates a brand new one with no children?